### PR TITLE
CI: e2e tests with cypress

### LINKED ISSUE #26: CI: e2e tests
We should add protractor tests, and check the new pages too

### PR DESCRIPTION
**Pull request by nassimerrahoui**
Fix #26

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Missing part in the CI
To be sure Cypress works well, in the CI, you need to :
- start the app (in background, to not block the output)
- launch Cypress

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "baseUrl": "http://localhost:9000",
```
Comment (pascalgrimaud) [inline cypress.json:2]: better to have 7419 instead of 9000 ?

File: .jhipster/modules/jhi-hooks.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[
```
Comment (pascalgrimaud) [inline .jhipster/modules/jhi-hooks.json:1]: Where does this file come from ? as I don't see an existing module for Cypress?

Comment (nassimerrahoui) [reply .jhipster/modules/jhi-hooks.json:1]: This file come from a custom module from https://github.com/clementdessoude/generator-jhipster-cypress but we change a lot of things here. So, we can delete this file as we will not use this custom module.

File: cypress.json (hunk 1 above)
Comment (nassimerrahoui) [reply cypress.json:2]: Yes, it is :)

File: package.json [hunk 3]
```diff
@@ -38,10 +38,13 @@
     "@vue/cli-service": "4.3.1",
     "@vue/test-utils": "1.0.2",
     "autoprefixer": "9.7.6",
+    "babel-core": "7.0.0-bridge.0",
     "browser-sync": "2.26.7",
     "browser-sync-webpack-plugin": "2.2.2",
     "copy-webpack-plugin": "6.0.0",
     "css-loader": "3.5.3",
+    "cypress": "4.3.0",
```
Comment (sendilkumarn) [inline package.json:46]: update to 4.8.0?

File: package.json [hunk 4]
```diff
@@ -118,7 +121,9 @@
     "webpack": "node --max_old_space_size=4096 node_modules/webpack/bin/webpack.js",
     "test": "npm run lint && npm run jest",
     "test-ci": "npm run lint && npm run jest:update",
-    "test:watch": "npm run jest -- --watch"
+    "test:watch": "npm run jest -- --watch",
```
Comment (sendilkumarn) [inline package.json:124]: To run cypress in the CI server, check this out https://docs.cypress.io/guides/guides/continuous-integration.html#Xvfb

File: src/test/javascript/cypress/support/authentication.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+Cypress.Commands.add('loginUsing', (username: string, password: string) => {
+  cy.server();
```
Comment (sendilkumarn) [inline src/test/javascript/cypress/support/authentication.ts:2]: This is nice, also check out [this](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=video&cd=&cad=rja&uact=8&ved=2ahUKEwiBs4Sxs5DqAhXRlqQKHVOtDacQtwIwA3oECAAQAQ&url=https%3A%2F%2Fdocs.cypress.io%2Fguides%2Fgetting-started%2Ftesting-your-app.html&usg=AOvVaw3W0X5-tml7M4CzJogRtX1i) video and [this](https://github.com/sendilkumarn/generator-jhipster/tree/feat/add-cypress) branch  (this kind of follows the structure mentioned by Cypress team).

**Comment by pascalgrimaud** [comment]
I'm in favor of merging this, but it is still in draft mode, that's why I didn't review it.
Is it ready @nassimerrahoui ?

**Comment by pascalgrimaud** [comment]
I'm convinced Cypress is our future ! ❤

**Comment by nassimerrahoui** [comment]
> I'm in favor of merging this, but it is still in draft mode, that's why I didn't review it.
> Is it ready @nassimerrahoui ?

This PR is working but we have another PR in vuejs blueprint with a different/better structure for Cypress. We want to apply the vuejs blueprint onto jhipster control center to update this PR before merge it. So, the code in this PR will change. WDYT ?

File: package.json (hunk 3 above)
Comment (nassimerrahoui) [reply package.json:46]: We have an issue when we want to update to a version of Cypress > 4.3.
There is an error about tslib.
cypress-error

Comment (nassimerrahoui) [reply package.json:46]: It seems that is a problem with multiple tsconfig.json (one for jhipster control center, one for cypress) but we still no found a solution for the moment.

**Comment by nassimerrahoui** [comment]
Hello, we worked on an overview of Cypress on this [](url)

**Comment by nassimerrahoui** [comment]
Closing this PR because we will use generator-jhipster to generate e2e tests with Cypress.

**Pull request closed without merging**